extensions/webchat/src/node/directline/store/DLServerState.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { BotEndpoint } from './entities/BotEndpoint';
+import { Attachments } from './entities/Attachments';
+import { ConversationSet } from './entities/ConversationSet';
+import { EndpointSet } from './entities/EndpointSet';
+import { LoggerLevel, LogItem } from './types';
+import { Conversation } from './entities/Conversation';
+
+export type DLServerState = {
+  conversations: ConversationSet;
+  endpoints: EndpointSet;
+  attachments: Attachments;
+  serviceUrl: string;
+  dispatchers: {
+    logToDocument: (
+      conversationId: string,
+      logMessage: LogItem<{
+        level: LoggerLevel;
+        text: string;
+      }>
+    ) => void;
+    getDefaultEndpoint: () => BotEndpoint;
+    updateConversation: (conversationId: string, updatedConversation: Conversation) => void;
+  };
+};
+
+class DLServerContext {
+  private static instance: DLServerContext;
+  public readonly state: DLServerState;
+
+  private constructor(serverPort?: number) {
+    this.state = {
+      conversations: new ConversationSet(),
+      endpoints: new EndpointSet(),
+      attachments: new Attachments(),
+      serviceUrl: serverPort ? `http://localhost:${serverPort}` : '',
+      dispatchers: {
+        logToDocument: this.logToDocument,
+        getDefaultEndpoint: this.getDefaultEndpoint,
+        updateConversation: this.updateConversation,
+      },
+    };
+  }
+
+  private logToDocument(
+    conversationId: string,
+    logItem: LogItem<{
+      level: LoggerLevel;
+      text: string;
+    }>
+  ) {
+    // TODO: Send the log item to the client Webchat instance to log errors.
+    // eslint-disable-next-line no-console
+    console.log(conversationId + logItem.payload.text);
```

Comment:
should we use `log` here instead?

Reply:
Using the debug package now @cwhitten . Essentially, in the next PR Im going to send the error messages through a socket to the debug panel in the client. Hence, this code will be replaced.